src/App.js

```diff
@@ -483,6 +499,7 @@ export class App extends React.Component<AppProps, AppState> {
                 break;
             case 'stopRequested': // Fall through
             case 'stopped':
+                this.handleClickStop();
```

Comment:
Case 'stopped' is when program is not running, so calling handleClickStop doesn't make sense over here, as what handleClickStop does is setting runningState to 'stopRequested'. Is this call intended?

Reply:
It doesn't seem appropriate, and the behaviour is as expected without it, so I removed it.